ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java

```diff
@@ -258,6 +259,39 @@ RaftGroup getGroup() {
     return RaftGroup.valueOf(getMemberId().getGroupId(), getRaftConf().getPeers());
   }
 
+  /**
+   * This removes the group from the server.
+   * If the flag is set to false, the directory is moved to
+   * {@link RaftServerConfigKeys#REMOVED_GROUPS_DIR_KEY} location only if
+   * {@link RaftServerConfigKeys#MOVE_REMOVED_GROUPS_ENABLED_KEY} is enabled.
+   * If the flag is true, the group is permanently deleted.
+   * @param deleteDirectory
+   */
+  public void groupRemove(boolean deleteDirectory) {
+    final RaftStorageDirectory dir = state.getStorage().getStorageDir();
+
+    /* Shutdown is triggered here inorder to avoid any locked files. */
+    shutdown(deleteDirectory);
+
+    if(!deleteDirectory && RaftServerConfigKeys.moveRemovedGroupsEnabled
```

Comment:
I think we can move the logic of delete directory from shutdown to this function.

Reply:
I have moved the logic to groupRemove